python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1693,49 +1694,67 @@ def create_run(
 
         return dagster_run
 
-    def _get_skipped_entity_keys(
-        self, run_id: str
+    def _get_keys_to_reexecute(
+        self, run_id: str, execution_plan_snapshot: "ExecutionPlanSnapshot"
     ) -> tuple[AbstractSet["AssetKey"], AbstractSet["AssetCheckKey"]]:
         """For a given run_id, return the set of asset keys and asset check keys that were planned but
-        not executed.
+        were not executed, or failed a blocking asset check.
         """
+        from dagster._core.definitions.asset_check_spec import AssetCheckKey
         from dagster._core.events import (
             AssetCheckEvaluation,
-            AssetCheckEvaluationPlanned,
-            AssetMaterializationPlannedData,
             DagsterEventType,
             StepMaterializationData,
         )
 
+        # Figure out the set of assets that were materialized and checks that successfully executed
         logs = self.all_logs(
             run_id=run_id,
             of_type={
-                DagsterEventType.ASSET_MATERIALIZATION_PLANNED,
                 DagsterEventType.ASSET_MATERIALIZATION,
-                DagsterEventType.ASSET_CHECK_EVALUATION_PLANNED,
                 DagsterEventType.ASSET_CHECK_EVALUATION,
             },
```

Comment:
i think this may need to query ASSET_MATERIALIZATION on its own separately in order to take advantage of the underlying optimizations that are possible

Reply:
my theory was that if we were already eating the scan for `ASSET_CHECK_EVALUATION`, it wouldn't be significantly more expensive to just lump the `ASSET_MATERIALIZATION` query in there as well vs. doing a query against the materializations table -- is that off base?

Reply:
I think it may be that querying for both event types separately may be faster than querying them together.  With multiple event types, it might just pick the `run_id, event_id` index and with a single event type, it might pick the `run_id, dagster_event_type, event_id` index.